lib/web_ui/lib/src/engine/bitmap_canvas.dart

```diff
@@ -489,10 +489,13 @@ class BitmapCanvas extends EngineCanvas {
   @override
   void drawParagraph(EngineParagraph paragraph, ui.Offset offset) {
     assert(paragraph._isLaidOut);
-    html.CanvasRenderingContext2D ctx = _canvasPool.context;
     final ParagraphGeometricStyle style = paragraph._geometricStyle;
 
     if (paragraph._drawOnCanvas && _childOverdraw == false) {
+      // !Do not move this assignment above this if clause since, accessing
+      // context will generate extra <canvas> tags.
+      html.CanvasRenderingContext2D ctx = _canvasPool.context;
```

Comment:
You can move it even lower, into `if (style != _cachedLastStyle)`, but not sure if it will make much difference.

Reply:
Done.